impl/src/metadata.rs

```diff
@@ -50,25 +97,132 @@ pub struct CargoWorkspaceFiles {
 /** A workspace metadata fetcher that uses the Cargo Metadata subcommand. */
 pub struct CargoMetadataFetcher {
   cargo_bin_path: PathBuf,
-  use_tempdir: bool,
+  registry_url: Url,
+  index_url: String,
 }
 
 impl CargoMetadataFetcher {
-  pub fn new<P: Into<PathBuf>>(cargo_bin_path: P, use_tempdir: bool) -> CargoMetadataFetcher {
+  pub fn new<P: Into<PathBuf>>(
+    cargo_bin_path: P,
+    registry_url: Url,
+    index_url: &str,
```

Comment:
Ignored this comment?

Reply:
Typically from my experience yes, `&String` would suggest that you are giving a reference directly to an owned thing, at which point since I dont get ownership of it in any meaningful way (`Box<String>` / `String` / `&mut String`) I am curious why not have the more primal type.

In the case of avoiding a copy but allowing it if needed I would more expect to see `Cow<'a, str>` as the type/

My two cents, I have not seen much `&String` in the wild

Reply:
I didn't mean &String, i meant String. Move the value.

The motivation here is basically that we already need an owned string here, but we take an &str because generally `&str` is what you want.

Here though that can result in an extra allocation if the end user no longer needed the string anyway.

Reply:
I've come across both cases. I'm not quite sure what all the trade-offs here are, ownership is the only one I've really had experience with. But I think you're right @acmcarther I think it'd be better to have the signature accept `String` to clearer communicate the intent and not mask any allocations.